compiler+runtime/src/cpp/jank/runtime/obj/tagged_literal.cpp

```diff
@@ -0,0 +1,99 @@
+#include <jank/runtime/obj/symbol.hpp>
+#include <jank/runtime/core/to_string.hpp>
+#include <jank/runtime/visit.hpp>
+#include <jank/runtime/context.hpp>
+#include <jank/runtime/obj/tagged_literal.hpp>
+
+namespace jank::runtime::obj
+{
+  tagged_literal::tagged_literal(object_ptr const tag, object_ptr const form)
+    : tag{ tag }
+    , form{ form }
+  {
+  }
+
+  native_bool tagged_literal::equal(object const &o) const
+  {
+    if(o.type != object_type::tagged_literal)
+    {
+      return false;
+    }
+
+    auto const s(expect_object<tagged_literal>(&o));
+    return tag == s->tag && form == s->form;
```

Comment:
This doesn't appear to be working:

```clojure
% ./build/jank repl             
Bottom of clojure.core
clojure.core=> (= (tagged-literal 'foo {}) (tagged-literal 'foo {}))
false
```

Reply:
By looking at similar code, I'm guessing `runtime::equal` is more appropriate to compare tag and form as they are not interned.

Reply:
Thanks @frenchy64, that was it! I don't know why I assumed that `==` would be overloaded instead of providing reference equality.